src/System.Management.Automation/engine/MshMemberInfo.cs

```diff
@@ -4825,37 +4904,163 @@ private PSMemberInfoInternalCollection<T> GetIntegratedMembers(MshMemberMatchOpt
             }
         }
 
-        /// <summary>
-        /// Returns all members in the collection matching name
-        /// </summary>
-        /// <param name="name">name of the members to be return. May contain wildcard characters.</param>
-        /// <returns>all members in the collection matching name</returns>
-        /// <exception cref="ArgumentException">for invalid arguments</exception>
-        public override ReadOnlyPSMemberInfoCollection<T> Match(string name)
+        private void GetIntegratedMembers(Func<T, bool> predicate, PSMemberInfoInternalCollection<T> result)
         {
-            if (string.IsNullOrEmpty(name))
+            using (PSObject.memberResolution.TraceScope("Generating the total list of members"))
             {
-                throw PSTraceSource.NewArgumentException("name");
+                object delegateOwner;
+
+                if (_mshOwner != null)
+                {
+                    delegateOwner = _mshOwner;
+                    foreach (PSMemberInfo member in _mshOwner.InstanceMembers)
+                    {
+                        if (member is T memberAsT && (predicate == null || predicate(memberAsT)))
+                        {
+                            result.Add(memberAsT);
+                        }
+                    }
+                }
+                else
+                {
+                    delegateOwner = _memberSetOwner.instance;
+                    foreach (PSMemberInfo member in _memberSetOwner.InternalMembers)
+                    {
+
+                        if (member is T memberAsT && (predicate == null || predicate(memberAsT)))
+                        {
+                            member.ReplicateInstance(delegateOwner);
+                            result.Add(memberAsT);
+                        }
+                    }
+                }
+
+                if (delegateOwner == null)
+                {
+                    return;
+                }
+
+                delegateOwner = PSObject.AsPSObject(delegateOwner);
```

Comment:
Seems we could use new var to exclude cast below in line 4947.
```suggestion
                var delegateOwnerAsPSObject = PSObject.AsPSObject(delegateOwner);
```